src/Http/Paginators/ResponsePaginator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Http\Paginators;
+
+use Generator;
+use ReturnTypeWillChange;
+use Saloon\Contracts\Response;
+use Saloon\Contracts\ResponsePaginator as ResponsePaginatorContract;
+use Saloon\Contracts\SerialisableResponsePaginator;
+
+use function count;
+
+/**
+ * @template TValue
+ *
+ * @implements ResponsePaginatorContract<TValue>
+ */
+abstract class ResponsePaginator implements ResponsePaginatorContract
+{
+    protected int $index = 1;
+
+    protected iterable $items;
+
+    public function __construct(
+        protected readonly Response $response,
+    ) {}
+
+    public function response(): Response
+    {
+        return $this->response;
+    }
+
+    /**
+     * @return iterable<int, TValue>
+     */
+    abstract protected function getItems(): iterable;
+
+    /**
+     * @return void
+     */
+    public function rewind(): void
+    {
+        $this->items = $this->getItems();
+    }
+
+    /**
+     * @return bool
+     */
+    public function valid(): bool
+    {
+        return $this->index < count($this->items);
+    }
+
+    /**
+     * @return TValue
+     */
+    #[ReturnTypeWillChange]
+    public function current(): mixed
+    {
+        return $this->internalIterator->current();
+    }
+
+    /**
+     * @return string|int
+     */
+    public function key(): string|int
+    {
+        return $this->internalIterator->key();
+    }
+
+    /**
+     * @return void
+     */
+    public function next(): void
+    {
+        $this->internalIterator->next();
+    }
+
+    /**
+     * @return void
+     */
+    public function previous(): void
+    {
+        $this->index--;
+    }
+}
```

Comment:
This looks interesting, although I can't see where it's being used

Reply:
https://github.com/juse-less/Saloon/commit/0aae49bacc4e5f2a4bdc9925bb40b74f00631033

There was already info on it, but added DocBlock examples.
For some reason PhpStorm won't show previews for me, so if someone is able to check out my branch and see so that the example renders correctly in DocBlocks, that'd be very helpful.

Reply:
Now that I think of it..
The `RequestPaginator::valid()` method always checks if the _next_ page is valid, not previous.
So this would cause issues.

I'd say it's highly unlikely anyone would use it anyway, so I'll just remove it altogether.
If a reverse iterator is needed, it can easily be created separately.

Reply:
I went through with it, and removed the `previous()` methods.